src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/service/GeneralPartnerService.java

```diff
@@ -60,7 +60,11 @@ public String createGeneralPartner(Transaction transaction, GeneralPartnerDto ge
         GeneralPartnerDao dao = mapper.dtoToDao(generalPartnerDto);
         GeneralPartnerDao insertedSubmission = insertDaoWithMetadata(requestId, transaction, userId, dao);
         String submissionUri = linkAndSaveDao(transaction, insertedSubmission.getId(), dao);
-        updateTransactionWithGeneralPartnerTransactionResourceLinks(requestId, transaction, submissionUri);
+        String generalPartnerId = dao.getId();
+        if (generalPartnerId == null) {
```

Comment:
I'm wondering what the point of this log line is? In theory, once partner has been saved on line 61 it will always have an id.

(The `dao` variable may not, as object not updated, but `insertedSubmission` will)